cmd/celestia-appd/cmd/testnet.go

```diff
@@ -0,0 +1,241 @@
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	"github.com/cosmos/cosmos-sdk/server"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/spf13/cast"
+	"github.com/spf13/cobra"
+
+	"github.com/celestiaorg/celestia-app/v4/app"
+)
+
+const valVotingPower int64 = 900000000000000
+
+var flagAccountsToFund = "accounts-to-fund"
+
+type valArgs struct {
+	newValAddr         bytes.HexBytes
+	newOperatorAddress string
+	newValPubKey       crypto.PubKey
+	accountsToFund     []string
+	upgradeToTrigger   string
+	homeDir            string
+}
+
+func NewInPlaceTestnetCmd() *cobra.Command {
+	cmd := server.InPlaceTestnetCreator(newTestnetApp)
+	cmd.Short = "Updates chain's application and consensus state with provided validator info and starts the node"
+	cmd.Long = `The test command modifies both application and consensus stores within a local mainnet node and starts the node,
+with the aim of facilitating testing procedures. This command replaces existing validator data with updated information,
+thereby removing the old validator set and introducing a new set suitable for local testing purposes. By altering the state extracted from the mainnet node,
+it enables developers to configure their local environments to reflect mainnet conditions more accurately.`
+	cmd.Example = `celestia-appd in-place-testnet testing-1 celestiavaloper1plclwg8j7lx42aufxmq5xsuuwddd0840f4a56d --home $HOME/.celestia-appd/validator1 --accounts-to-fund="celestia1hhtl7m03qnkk9s0ane7dr720ujegq03sd64arq,celestia1plclwg8j7lx42aufxmq5xsuuwddd0840v2ldvt"`
+
+	cmd.Flags().String(flagAccountsToFund, "", "Comma-separated list of account addresses that will be funded for testing purposes")
+	return cmd
+}
+
+// newTestnetApp starts by running the normal newApp method. From there, the app interface returned is modified in order
+// for a testnet to be created from the provided app.
+func newTestnetApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts servertypes.AppOptions) servertypes.Application {
+	// Create an app and type cast to an App
+	newApp := NewAppServer(logger, db, traceStore, appOpts)
+	testApp, ok := newApp.(*app.App)
+	if !ok {
+		panic("app created from newApp is not of type App")
+	}
+
+	// Get command args
+	args, err := getCommandArgs(appOpts)
+	if err != nil {
+		panic(err)
+	}
+
+	return initAppForTestnet(testApp, args)
+}
+
+func initAppForTestnet(app *app.App, args valArgs) *app.App {
+	// Required Changes:
+	//
```

Comment:
These comments seem strangely formatted

```suggestion
```

Reply:
agree, they were pulled in from sdk's testnet cmd I believe